src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/Extensions/SyntaxGeneratorExtensions_Negate.cs

```diff
@@ -48,6 +48,18 @@ internal static partial class SyntaxGeneratorExtensions
             SemanticModel semanticModel,
             bool negateBinary,
             CancellationToken cancellationToken)
+        {
+            return Negate(generator, generatorInternal, expressionOrPattern, semanticModel, negateBinary, swapBooleans: true, cancellationToken);
```

Comment:
I'm curious why we're always passing true here, instead of basing it off of the expression being negated.  Could we do that?  If not, could you doc the reasoning here?  Thanks!

Reply:
The idea was just to keep the existing behavior, except for calls from `GetNegationOfIsPatternExpression` where we know whether swapping `true` and `false` is valid or not based on the value type in the is pattern. I don't really know how we would base the `swapBooleans` argument here on the expression, since the analysis of the expression that determines that happens inside the `Negate` method we are calling. Maybe I'm misunderstanding you?

An alternative could be to remove the `swapBooleans` parameter again and just check for constant patterns in `GetNegationOfIsPatternExpression` and directly call `GetNegationOfConstantPattern` with `swapBooleans: true`. I'll just look into whether a similar problem exists for inequality patterns, as suggested by @jnm2 [here](https://github.com/dotnet/roslyn/issues/64292#issuecomment-1269181687), and how that affects this code.

Reply:
On a related note, if we keep this parameter, maybe it should be called `allowSwappingBooleans` instead.

Reply:
i think i get it :)  this seems reasonable.

> aybe it should be called allowSwappingBooleans instead.

i like that :)   could you make that change?  i'll sign off and merge then :)

Reply:
Thanks! Done.